ftcosi/protocol/sub_protocol.go

```diff
@@ -260,6 +259,19 @@ func (p *SubFtCosi) Dispatch() error {
 		p.subResponse <- responses[0]
 	} else {
 		// generate own response and send to parent
+		if challenge.AggregateCommit == nil || challenge.Mask == nil {
+			log.Warn("Only have pre-calculated challange - this is dangerous!")
+		} else {
+			cosiChallenge, err := cosi.Challenge(p.suite, challenge.AggregateCommit,
+				challenge.Mask.AggregatePublic, p.Msg)
+			if err != nil {
+				return err
+			}
+			if !cosiChallenge.Equal(challenge.CoSiChallenge) {
```

Comment:
I was not expecting this branch of the code to read from challenge.CoSiChallenge, since it is deprecated. I expected the output of cosi.Challenge to be used directly.

Reply:
Not sure I can do better and easier to provide both backward-compatibility and warning about bad behaviour. Any proposition how to do better here?

Reply:
I guess what I mean is that this code is checking a value that the docs say should no longer be sent. So when it is not sent, this code will fail. If the new format (i.e. provide Agg and Mask) is provided, why not just use it 100% and ignore the old format?

Reply:
I think it must be sent because some nodes might not be on the latest code.

Reply:
Ah, this is what I was missing, that gnerateResponse will be using the challenge nextly. OK, got it, kind of.